observer/holder/nodesHolder.go

```diff
@@ -0,0 +1,436 @@
+package holder
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"sync"
+
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-chain-proxy-go/data"
+)
+
+var (
+	errEmptyShardIDsList  = errors.New("empty shard IDs list")
+	errWrongConfiguration = errors.New("wrong observers configuration")
+	log                   = logger.GetOrCreate("observer/holder")
+)
+
+type nodesHolder struct {
+	mut                    sync.RWMutex
```

Comment:
now with this new mut we'll have lock in lock

Reply:
refactored